esp/esp/program/modules/handlers/nametagmodule.py

```diff
@@ -107,10 +111,45 @@ def generatetags(self, request, tl, one, two, module, extra, prog):
                               'id'   : teacher.id,
                               'username': teacher.username})
 
+        elif idtype == 'other':
+            users_list = []
+            if request.POST['group'] == '':
+                raise ESPError("You need to select a group", log=False)
+            group = request.POST['group']
+            user_Q = Q(groups=group)
+            users_list = ESPUser.objects.filter(user_Q).distinct()
+
+            users_list = [ user for user in users_list ]
+            users_list = filter(lambda x: len(x.first_name+x.last_name), users_list)
+            users_list.sort()
+
+            user_title = request.POST['blanktitle']
+            for user in users_list:
+                users.append({'title': user_title,
+                              'name' : '%s %s' % (user.first_name, user.last_name),
+                              'id'   : user.id,
+                              'username': user.username})
+
         elif idtype == 'volunteers':
```

Comment:
Can we duplicate a bit less of the logic between user types here?  You can just do the if-else block for generating the list of users (and title), and then computing the actual nametag data from there is almost the same in each case (except misc/blank).  I know we were already duplicating code, but since you're adding two more copies it's a good time to do a little refactoring.